src/fetchWrapper.js

```diff
@@ -13,6 +15,7 @@ export default function fetchWrapper(reqFromRelay, middlewares) {
     }
 
     return fetch(url, opts)
+      .then(response => throwOnServerError(request, response))
```

Comment:
I just realized throwing from here breaks the retry middleware.
Too bad there were no tests yet.
I'll try to fix that. So don't merge yet ;)